src/Microsoft.Owin/Extensions/AppBuilderCookieExtensions.cs

```diff
@@ -0,0 +1,111 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.Owin.Infrastructure;
+using Owin;
+
+namespace Microsoft.Owin
+{
+    /// <summary>
+    /// Cookie-related extensions for <see cref="IAppBuilder"/>.
+    /// </summary>
+    public static class AppBuilderCookieExtensions
+    {
+        /// <summary>
+        /// Sets the default <see cref="ICookieManager"/> that will be used by
+        /// the OWIN middleware when no manager is explicitly set in their options.
+        /// </summary>
+        /// <remarks>
+        /// Depending on the host, a default <see cref="ICookieManager"/> instance
+        /// may already be registered in <see cref="IAppBuilder.Properties"/>
+        /// (e.g SystemWeb automatically registers a cookie manager that relies on
+        /// HttpCookieCollection instead of directly manipulating the HTTP headers).
+        /// </remarks>
+        /// <param name="app">The application builder.</param>
+        /// <param name="manager">The cookie manager.</param>
+        /// <exception cref="ArgumentNullException"><paramref name="app"/> is <see langword="null"/>.</exception>
+        public static void SetDefaultCookieManager(this IAppBuilder app, ICookieManager manager)
+        {
+            if (app is null)
+            {
+                throw new ArgumentNullException(nameof(app));
+            }
+
+            if (manager is null)
+            {
+                app.Properties.Remove("infrastructure.CookieManager");
+            }
+            else
+            {
+                app.Properties["infrastructure.CookieManager"] = manager;
+            }
+        }
+
+        /// <summary>
+        /// Sets the default chunking <see cref="ICookieManager"/> that will be used
+        /// by the OWIN middleware when no manager is explicitly set in their options.
+        /// </summary>
+        /// <remarks>
+        /// Depending on the host, a default <see cref="ICookieManager"/> instance
+        /// may already be registered in <see cref="IAppBuilder.Properties"/>
+        /// (e.g SystemWeb automatically registers a cookie manager that relies on
+        /// HttpCookieCollection instead of directly manipulating the HTTP headers).
+        /// </remarks>
+        /// <param name="app">The application builder.</param>
+        /// <param name="manager">The cookie manager.</param>
+        /// <exception cref="ArgumentNullException"><paramref name="app"/> is <see langword="null"/>.</exception>
+        public static void SetDefaultChunkingCookieManager(this IAppBuilder app, ICookieManager manager)
```

Comment:
Should we add type checks to these extensions to detect rookie mistakes, like calling `SetDefaultCookieManager` with a `ChunkingCookieManager` instance or `SetDefaultChunkingCookieManager` with a `CookieManager`? (it wouldn't work with a simple `typeof(...)` with the SystemWeb managers as they are in another assembly, or we would need to check using the name)

Reply:
No, these are general helpers, they don't decide how they're used. I could see these used for the SameSiteCookieManager's.